shell/components/nav/Type.vue

```diff
@@ -142,63 +79,70 @@ export default {
+    v-slot="{ href, navigate, isActive, isExactActive }"
+    custom
     :to="type.route"
-    tag="li"
-    class="child nav-type"
-    :class="{'root': isRoot, [`depth-${depth}`]: true, 'router-link-active': isCurrent}"
     :exact="type.exact"
   >
-    <TabTitle
-      v-if="$router.resolve(type.route).route.path === $route.path"
-      :show-child="false"
-    >
-      {{ type.labelKey ? t(type.labelKey) : (type.labelDisplay || type.label) }}
-    </TabTitle>
-    <a
-      @click="selectType"
-      @mouseenter="setNear(true)"
-      @mouseleave="setNear(false)"
+    <li
+      class="child nav-type"
+      :class="{'root': isRoot, [`depth-${depth}`]: true, 'router-link-active': isActive, 'router-link-exact-active': isExactActive}"
+      @click="navigate"
+      @keypress.enter="navigate"
     >
-      <span
-        v-if="type.labelKey"
-        class="label"
-      ><t :k="type.labelKey" /></span>
-      <span
-        v-else
-        v-clean-html="type.labelDisplay || type.label"
-        class="label"
-        :class="{'no-icon': !type.icon}"
-      />
-      <span
-        v-if="showFavorite || namespaceIcon || showCount"
-        class="count"
+      <TabTitle
+        v-if="isExactActive"
+        :show-child="false"
       >
-        <Favorite
-          v-if="showFavorite"
-          :resource="type.name"
-        />
-        <i
-          v-if="namespaceIcon"
-          class="icon icon-namespace"
-          :class="{'ns-and-icon': showCount}"
+        {{ type.labelKey ? t(type.labelKey) : (type.labelDisplay || type.label) }}
+      </TabTitle>
+      <a
+        :href="href"
+        @click="selectType"
+        @mouseenter="setNear(true)"
+        @mouseleave="setNear(false)"
+      >
+        <span
+          v-if="type.labelKey"
+          class="label"
+        ><t :k="type.labelKey" /></span>
+        <span
+          v-else
+          v-clean-html="type.labelDisplay || type.label"
+          class="label"
+          :class="{'no-icon': !type.icon}"
         />
         <span
-          v-if="showCount"
-        >{{ count }}</span>
-      </span>
-    </a>
+          v-if="showFavorite || namespaceIcon || showCount"
+          class="count"
+        >
+          <Favorite
+            v-if="showFavorite"
+            :resource="type.name"
+          />
+          <i
+            v-if="namespaceIcon"
+            class="icon icon-namespace"
+            :class="{'ns-and-icon': showCount}"
+            data-testid="type-namespaced"
+          />
+          <span
+            v-if="showCount"
+            data-testid="type-count"
+          >{{ count }}</span>
+        </span>
+      </a>
+    </li>
   </router-link>
   <li
     v-else-if="type.link"
     class="child nav-type"
+    data-testid="link-type"
   >
     <a
       :href="type.link"
       :target="type.target"
       rel="noopener noreferrer nofollow"
-      @click="selectType"
-      @mouseenter="setNear(true)"
-      @mouseleave="setNear(false)"
```

Comment:
- `selectType` is specific to types in our app while this is being treated as a link to an external page based on the `rel=".."`.
- `setNear` is only actually used in relation to showing the Favorite component which isn't possible in this link.